man/systemd.network.xml

```diff
@@ -1759,6 +1759,17 @@
           </listitem>
         </varlistentry>
 
+        <varlistentry>
+          <term><varname>AssignAcquiredDelegatedPrefixAddress=</varname></term>
+          <listitem>
+            <para>Takes a boolean. In DHCPv6-PD environment, where WAN interface requests IPv6 via DHCPv6,
+            receives address as well as delegated prefixes, with LAN interfaces serving those delegated
+            prefixes in their router advertisement messages. When true (on LAN interface), adds an address to
+            LAN interface from the delegated prefixes which are received from the WAN interface using EUI-64
+            algorithm. Default to true.</para>
```

Comment:
EUI-64 describes only how to create the last 64 bits of an IP address. I think the more interesting bit here is how the 64 bits before the EUI-64 part are assembled, which this doesn't explain at all.

Reply:
Can you suggest ?

Reply:
@flokli it will be from the prefix that has been selected for the given interface.

https://github.com/systemd/systemd/pull/15259/files#diff-30d862b9251b785c8ee19db9d7a0321aR70-R114

Networkd already assigns one /64 to each interface that has `IPv6PrefixDelegation=dhcpv6` configured. The very same prefix will be used for the EUI-64 address.

Reply:
@flokli While I agree the mechanics of picking the /64 prefix to delegate to a subnet (or subnets) is not explained anywhere in the manpages, adding this explaination, IMHO, is out of scope for this change. This fix is for very specific issue of "we already have an interface serving the subnet, that subnet already has /64 delegated and is serving RAs and forwarding traffic, but lacks address on the interface *from that same prefix*".

To illustrate why this is wrong place for this explanation, consider that you will need the same explanation for "what prefix is RA going to be sent for", even if you never specify this AssignAcquiredDelegatedPrefix option.

Describing the mechanics of assembling /64 prefix for subnet assignment from /48, /56, /60 delegated from upstream DHCP is better handled in Network.IPv6PrefixDelegation documentation. That definitely needs some work - but let's not scope creep this fix please.

Reply:
Nah, I didn't meant to scope creep. Ack on not describing the whole concept in that option, but we should refer to the place where it will be (hopefully soon better) documented.

What about something like that:

```
<listitem><para>Whether to add an address from the prefix assigned
to the interface by the<varname>IPv6PrefixDelegation=</varname>
option.
Will synthesize the interface identifier via the EUI-64 algorithm[, or
via <varname>IPv6Token=</varname> if set.] Defaults to true.
Only effective if <varname>IPv6PrefixDelegation=</varname>
isn't false.</para></listitem>
```

Reply:
`IPv6Token=`  is not implemented here .

Reply:
yeah, which is why this was in square brackets ;-)